docs/building/forms.md

```diff
@@ -0,0 +1,107 @@
+
+The user will be taken through a wizard and it will look like:
+
+```
+#TODO CREATE TASK
+Provide a description
+```
+
+This is the first step and whatever the user replies with will be set as the task description. This is a **string** and it will be **POST**ed to **path** as indicated in the json structure. Notice there is no **header** or **footer** mentioned at this step, so form header and footer will be used as fallbacks.
+
+
+```
+#TODO DUE DATE
+Provide a description
+```
+
+This is the second step and the user needs to reply with a date. There is no callback path so no HTTP request will be performed after this step. The **header** is present here, so it will be displayed instead of the form header.
+
+
+```
+#TODO PRIORITY
+A High priority
+B Low priority
+--Reply A-B
+```
+
+This is the last step and the user needs to choose one option.
+
+After all steps have been processed, as indicated through **method** and **path**, an HTTP POST will be sent to the path which is relative to the callback url: **http://your-callback.url/task/create/** under the mapping **form_item_name=user_choice**
+
+So the POST will look like: **?name=some_description&date=2019-10-10&prio=high**
+
+
+## Type
+The response **type** should be equal to **form** to indicate a form response.
+
+## Meta
+The **meta** key holds a dictionary which can contain one of the following keys:
+
+- **confirmation_needed** - if set to false, there won't be any confirmation screen at the end of the form (defaults to true)
+- **completion_status_show** - if set to true, there will be a completion status shown (defaults to false)
+- **completion_status_in_header** - if set to true, the status will be shown in the header (defaults to false, applicable only when status is shown)
+
+## Header
+The header of the form is indicated through **header** key. This value is not final and will be altered by the ONEm platform, by making it uppercased and placing the name of your app in front of it.
+
+
+## Footer
+The footer of the form is indicated through **footer** key and like the header of the form, it is not final and will be altered by the ONEm platform. If no footer is specified, a default might be set.
+
+## Body
+The body holds a sequence of dictionaries. Each dictoinary is called a form item and it is described through the below keys:
+
+- **type** - the form item type: _string, int, float, date, datetime, menu_
+- **name** - the form item name - this is an identifier and it is used in the information collection process
+- **description** - the form item description
+- **path** - if exists, an HTTP request will be made after the user submitted his choice
+- **method** - indicates how the HTTP request will be made (GET, POST, etc...)
+- **header** - indicates the header for this form item and will overwrite the form **header**
+- **footer** - indicates the footer for this form item and will overwrite the form **footer**
```

Comment:
You forgot the `body` field here (it is present in the example above).

Reply:
The body field in a form item refers to a form item menu. See below

Reply:
Yeah, but `type` is `menu` for both. So they should have the same structure or their type should be different.

Reply:
Its a type menu as a form item, not on the root level. If you think this creates confusion we could change it to something like `form_menu` ?